lua/astrocommunity/editing-support/vim-visual-multi/init.lua

```diff
@@ -0,0 +1,33 @@
+return {
+  "mg979/vim-visual-multi",
+  event = { "User AstroFile", "InsertEnter" },
+  dependencies = {
+    "astronvim/astrocore",
+    ---@param opts astrocoreopts
+    opts = function(_, opts)
+      if not opts.options then opts.options = {} end
+      if not opts.options.g then opts.options.g = {} end
+      if not opts.options.opt then opts.options.opt = {} end
+      opts.options.g.VM_silent_exit = 1
+      opts.options.g.VM_show_warnings = 0
+
+      if not opts.autocmds then opts.autocmds = {} end
+      opts.autocmds.visual_multi_exit = {
+        {
+          event = { "User" },
+          pattern = "visual_multi_exit",
+          desc = "Avoid spurious 'hit-enter-prompt' when exiting vim-visual-multi",
+          callback = function()
+            vim.o.cmdheight = 1
+            vim.schedule(function() vim.o.cmdheight = opts.options.opt.cmdheight end)
+          end,
```

Comment:
I'll just put it here that I'm extremely not a fan of this approach because this is a pretty brutal hack on how scheduling functions work and doesn't necessarily guarantee that the scheduled function will happen at the time you want it to. Instead this issue should be run upstream rather than a hacky fix here. If the user wants to avoid this issue then they can just set `cmdheight=1` in their config

Reply:
that being said, it's up to you if you want to keep it here :joy: I don't care what ends up in AstroCommunity

Reply:
I agree – then again, we don't care when it happens. We just need to quickly set cmdheight to 1 on the exit event and flip it back to 0 whenever. If my understanding stand, flipping back to 0 is scheduled to run in the next event loop, which will be executed after the current one (where exit is at, assuming sequential order) has finished, so timing issues shouldn't happen.

Let's keep it as it is for now. I'll raise an issue upstream, and we can remove this once we have a proper fix in our hands.